btf/types.go

```diff
@@ -1033,11 +1056,11 @@ func inflateRawTypes(rawTypes []rawType, rawStrings *stringTable) ([]Type, error
 
 	for _, fixup := range fixups {
 		i := int(fixup.id)
-		if i >= len(types) {
+		if i >= len(types)+len(baseTypes) {
 			return nil, fmt.Errorf("reference to invalid type id: %d", fixup.id)
 		}
 
-		*fixup.typ = types[i]
+		*fixup.typ = types[i-len(baseTypes)]
 	}
 
 	for _, bitfieldFixup := range bitfieldFixups {
```

Comment:
This PR doesn't address legacy bitfield fixups at all, which needs another look. Basically, what if a struct member in a kmod points at a type in base which used a legacy bitfield encoding? At this point we've lost the bitfield information from `base`, so there is probably nothing we can do. Except return an error.

Reply:
> At this point we've lost the bitfield information from base, so there is probably nothing we can do

Also, the new bitfield format was introduced in v4.20 (https://github.com/torvalds/linux/commit/9d5f9f701b1891466fb3dbb1806ad97716f95cc3), while the kmod BTF - in v5.9 (https://github.com/torvalds/linux/commit/36e68442d1afd4f720704ee1ea8486331507e834).

> Except return an error.

Added the check.